### PR TITLE
mofem: add new releases and minor changes

### PR DESCRIPTION
**Pull request by likask**
- Add new releases
- Add versions to mofem users modules packages and dependencies
- Bump maximal dependency to PETSc 
- Add SLEPC varianat
- Fix spelling bugs

### CODE REVIEW

File: var/spack/repos/builtin/packages/moab/package.py [hunk 1]
```diff
@@ -38,6 +38,8 @@ class Moab(AutotoolsPackage):
     homepage = "https://bitbucket.org/fathomteam/moab"
     url = "http://ftp.mcs.anl.gov/pub/fathom/moab-5.0.0.tar.gz"
 
+    version('5.0.2', '00a6f96f2e6591ab087548839fa3825e',
+             url='ftp://ftp.mcs.anl.gov/pub/fathom/moab-5.0.2.tar.gz')
```
Comment (davydden) [inline var/spack/repos/builtin/packages/moab/package.py]: you should not need to specify `url`, should be picked up by Spack just fine.

Comment (davydden) [reply var/spack/repos/builtin/packages/moab/package.py]:
```
version('5.0.2', '00a6f96f2e6591ab087548839fa3825e')
```
should be enough, Spack should figure out the right `url` from ` url = "http://ftp.mcs.anl.gov/pub/fathom/moab-5.0.0.tar.gz"`.

Comment (likask) [reply var/spack/repos/builtin/packages/moab/package.py]: Thanks. I've tested this, it works. Spack magically found right tarball.

File: var/spack/repos/builtin/packages/mofem-cephas/package.py [hunk 2]
```diff
@@ -24,15 +24,28 @@
 
 
 class MofemCephas(CMakePackage):
-    """mofem-cephas core library"""
+    """MoFEM is finite element core library"""
 
     homepage = "http://mofem.eng.gla.ac.uk"
-    git      = "https://bitbucket.org/likask/mofem-cephas.git"
+    url = "https://bitbucket.org/likask/mofem-cephas.git"
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/mofem-cephas/package.py]: Can you undo this change? This was likely due to a bad rebase. Spack now allows packages to specify a package-level `git` attribute so that you don't need to specify the same `git` for every version.

File: var/spack/repos/builtin/packages/mofem-fracture-module/package.py [hunk 3]
```diff
@@ -27,12 +27,28 @@ class MofemFractureModule(CMakePackage):
     """mofem fracture module"""
 
     homepage = "http://mofem.eng.gla.ac.uk"
-    git = "https://bitbucket.org/likask/mofem_um_fracture_mechanics.git"
+    url = "https://bitbucket.org/likask/mofem_um_fracture_mechanics"
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/mofem-fracture-module/package.py]: Can you undo this?

File: var/spack/repos/builtin/packages/mofem-users-modules/package.py [hunk 4]
```diff
@@ -43,14 +43,28 @@ class MofemUsersModules(CMakePackage):
 
     homepage = "http://mofem.eng.gla.ac.uk"
     url = "https://bitbucket.org/likask/mofem-joseph/downloads/users_modules_dummy"
-    version('1.0', '5a8b22c9cdcad7bbad92b1590d55edb1', expand=False)
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/mofem-users-modules/package.py]: What happened to this release?

Comment (likask) [reply var/spack/repos/builtin/packages/mofem-users-modules/package.py]: In previous version extension, 1.0 worked with all versions of the core library.  However, packages are better to manage if the extension version, i.e. mofem-usres-modules, follows package mofem-cpehas.  

I removed the old version, since ad hoc I put a version of it as 1.0. If it will be left it will be the higher version any other. So spack will install version 1.0 all the times when someone does
spack install mofem-users-modules

Instead of installing the most recent, in this case, mofem-users-modules@0.8.12.

File: var/spack/repos/builtin/packages/mofem-fracture-module/package.py (hunk 3 above)
Comment (likask) [reply var/spack/repos/builtin/packages/mofem-fracture-module/package.py]: Ok. Done.

**Comment by likask** [comment]
Hello, @adamjstewart @davydden, can you look at this pull request.

File: var/spack/repos/builtin/packages/mofem-users-modules/package.py [hunk 5]
```diff
@@ -43,14 +43,28 @@ class MofemUsersModules(CMakePackage):
...
+
+    version('0.8.12', '5a8b22c9cdcad7bbad92b1590d55edb1', expand=False)
```
Comment (davydden) [inline var/spack/repos/builtin/packages/mofem-users-modules/package.py]: Frankly, I am not a fan of these dummy versions that duplicate the main dependency. 

If users need to control which mofem-cephas to use, this shall be done on CLI or in packages.yaml

Comment (likask) [reply var/spack/repos/builtin/packages/mofem-users-modules/package.py]: Users modules is a git submodule of mofem core lib, it is not a dummy in the sense that which each version different code of user module is compiled. 

Currently is a bit of lack of consistency form my side, since the source is installed with git submodules with core lib, and package mofem-users-modules is used to compile this source code. 

I going to move correctly this with next pull request, that mofem-users-modules will work as a proper extension. However, I need to have this transition stage.

Comment (davydden) [reply var/spack/repos/builtin/packages/mofem-users-modules/package.py]: Ok, sounds good to me

**Comment by likask** [comment]
@davydden @adamjstewart I decided to speed up changes, have look to recent updates. Now mofem-users-modules are installed for git repo into native spack source directory. From another hand, git submodule is not cloned with core lib. At that point is the standard case for an extension, without any bending of spack rules.

File: var/spack/repos/builtin/packages/mofem-users-modules/package.py [hunk 6]
```diff
@@ -25,44 +25,36 @@
...
-    view symlink um_view mofem-cephas, and activate the extension, i.e. spack
-    activate um_view mofem-minimal-surface-equation. Basic mofem
-    functionality is available when with spack install mofem-users-modules,
-    it provides simple examples for calculating elasticity problems,
-    magnetostatics, saturated and unsaturated flow and a couple more. For
-    more information how to work with Spack and MoFEM see
-    http://mofem.eng.gla.ac.uk/mofem/html/install_spack.html"""
+    modules and extends of mofem-cephas package. For more information how to
+    work with Spack and MoFEM see
+    http: // mofem.eng.gla.ac.uk / mofem / html / install_spack.html """
```
[the remark this answers is not in the document]
Comment (likask) [reply var/spack/repos/builtin/packages/mofem-users-modules/package.py]: It does not. Thx. I will fix this.

File: var/spack/repos/builtin/packages/mofem-cephas/package.py [hunk 7]
```diff
@@ -24,15 +24,34 @@
...
+        tag='v0.8.10')
+    version('0.8.9', git='https://bitbucket.org/likask/mofem-cephas.git',
+        tag='v0.8.9')
+    version('0.8.8', git='https://bitbucket.org/likask/mofem-cephas.git',
+        tag='v0.8.8')
+    version('0.8.7', git='https://bitbucket.org/likask/mofem-cephas.git',
+        tag='v0.8.7')
+    version('develop',
+        git='https://bitbucket.org/likask/mofem-cephas.git',
+        branch='develop')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/mofem-cephas/package.py]: Since you have a package-level `git` variable, you no longer need to specify `git` for each version. Same for most of your other packages.

Comment (likask) [reply var/spack/repos/builtin/packages/mofem-cephas/package.py]: Fixed.

File: var/spack/repos/builtin/packages/mofem-cephas/package.py [hunk 8]
```diff
@@ -24,31 +24,41 @@
...
+    version('0.8.15', tag='v0.8.15')
+    version('0.8.14', tag='v0.8.14')
+    version('0.8.13', tag='v0.8.13')
+    version('0.8.12', tag='v0.8.12')
+    version('0.8.11', tag='v0.8.11')
+    version('0.8.10', tag='v0.8.10')
+    version('0.8.9', tag='v0.8.9')
+    version('0.8.8', tag='v0.8.8')
+    version('0.8.7', tag='v0.8.7')
+    version('develop', branch='develop')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/mofem-cephas/package.py]: These no longer need `submodules=True`?

File: var/spack/repos/builtin/packages/mofem-fracture-module/package.py [hunk 9]
```diff
@@ -31,13 +31,23 @@ class MofemFractureModule(CMakePackage):
...
-    version('develop', branch='develop')
+    version('0.9.50', tag='v0.9.50')
+    version('0.9.49', tag='v0.9.49')
+    version('0.9.48', tag='v0.9.48')
+    version('0.9.47', tag='v0.9.47')
+    version('0.9.46', tag='v0.9.46')
+    version('0.9.45', tag='v0.9.45')
+    version('0.9.44', tag='v0.9.44')
     version('0.9.42', tag='v0.9.42')
+    version('develop', branch='develop')
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/mofem-fracture-module/package.py]: I prefer to put `develop` at the top since it's the newest version, but that's not a hard requirement.

Comment (likask) [reply var/spack/repos/builtin/packages/mofem-fracture-module/package.py]: Good point, fix on the way.

**Comment by alalazo** [comment]
Thanks!
